lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1297,11 +1297,7 @@ public function validateIdentifier()
     public function validateAssociations()
     {
         foreach ($this->associationMappings as $mapping) {
-            if (
-                ! class_exists($mapping['targetEntity'])
-                && ! interface_exists($mapping['targetEntity'])
-                && ! trait_exists($mapping['targetEntity'])
-            ) {
+            if (! class_exists($mapping['targetEntity']) && ! interface_exists($mapping['targetEntity'])) {
```

Comment:
Thinking a bit more about this, I don't think we want to allow interfaces either?

This method is called at a very late stage of loading metadata, and nothing is going to fixup/fill/amend metadata after that; and the result is meant to be read-only.

So, how can we deal with an interface name as the association target?

If `ResolveTargetEntityListener` comes to mind, that should have taken place before we run this check.